osu.Game/Rulesets/Objects/HitObject.cs

```diff
@@ -72,18 +69,16 @@ public void ApplyDefaults(ControlPointInfo controlPointInfo, BeatmapDifficulty d
         {
             ApplyDefaultsToSelf(controlPointInfo, difficulty);
 
-            if (nestedHitObjects.IsValueCreated)
-                nestedHitObjects.Value.Clear();
+            nestedHitObjects.Clear();
 
             CreateNestedHitObjects();
 
-            if (nestedHitObjects.IsValueCreated)
+            nestedHitObjects.Sort((h1, h2) => h1.StartTime.CompareTo(h2.StartTime));
```

Comment:
Either change back to `SortedList` as discussed, or add a comment explaining why we're using post-sort.